`make lint` fails to install the golangci-lint binary

When running as a regular user the install of the golangci-lint fails due to permission denied.

```
make lint
if ! command -V golangci-lint; then curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b /bin v1.62.2 ; fi
/bin/sh: line 1: command: golangci-lint: not found
golangci/golangci-lint info checking GitHub for tag 'v1.62.2'
golangci/golangci-lint info found version: 1.62.2 for v1.62.2/linux/amd64
install: cannot create regular file '/bin/golangci-lint': Permission denied
make: *** [Makefile:50: install-metrics-binaries] Error 1
```

https://github.com/kubevirt/project-infra/blob/739601a1aed85f1139fa5829bac6b00e652e298d/Makefile#L50

This install should probably be removed and we should try to use something like a prebuilt container with podman.

--- a/Makefile
+++ b/Makefile
@@ -46,10 +46,7 @@ test: build
 update-labels:
 	./hack/labels/update.sh
 
-install-metrics-binaries:
-	if ! command -V golangci-lint; then curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b ${GOPATH}/bin ${GOLANGCI_LINT_VERSION} ; fi
-
-lint: install-metrics-binaries
+lint:
 	./hack/lint.sh
 
 coverage:

--- a/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
+++ b/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
@@ -110,11 +110,10 @@ presubmits:
             memory: "4Gi"
         securityContext:
           runAsUser: 0
-  - always_run: false
+  - always_run: true
     cluster: kubevirt-prow-control-plane
     decorate: true
     name: pull-project-infra-lint
-    run_if_changed: "external-plugins/.*|releng/.*|robots/.*"
     spec:
       containers:
       - command:

--- a/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
+++ b/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml
@@ -117,12 +117,18 @@ presubmits:
     run_if_changed: "external-plugins/.*|releng/.*|robots/.*"
     spec:
       containers:
-      - args:
-        - make lint
-        command:
+      - command:
         - "/usr/local/bin/runner.sh"
         - "/bin/sh"
         - "-ce"
+        - |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b ${GOPATH}/bin ${GOLANGCI_LINT_VERSION}
+          make lint
+        env:
+        - name: GIMME_GO_VERSION
+          value: "1.24.1"
+        - name: GOLANGCI_LINT_VERSION
+          value: "v1.64.8"
         image: quay.io/kubevirtci/golang:v20250327-c3122d8
         resources:
           requests:

--- a/go.mod
+++ b/go.mod
@@ -232,5 +232,6 @@ replace (
 	k8s.io/client-go => k8s.io/client-go v0.30.1
 )
 
-go 1.23
+go 1.23.0
+
 toolchain go1.24.1

--- a/hack/lint.sh
+++ b/hack/lint.sh
@@ -22,6 +22,10 @@ set -e
 set -u
 set -o pipefail
 
+if ! command -V golangci-lint; then
+    echo "golangci-lint binary required to be installed locally" && exit 1;
+fi
+
 golangci-lint --version
 golangci-lint run
 

